internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +469,67 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithValidObject() {
```

Comment:
what happens if sources parameter is nil?

Reply:
added testcase for the same